src/Interpreters/DeadLetter.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+#include <variant>
+#include <Interpreters/SystemLog.h>
+#include <Core/NamesAndTypes.h>
+#include <Core/NamesAndAliases.h>
+#include <Storages/ColumnsDescription.h>
+
+namespace DB
+{
+
+struct DeadLetterElement
+{
+    enum class StreamType : int8_t
+    {
+        Kafka = 1,
+        RabbitMQ = 2,
+    };
+
+    StreamType table_engine;
+    UInt64 event_time{};
+    Decimal64 event_time_microseconds{};
+
+    String database;
+    String table;
+    String raw_message;
+    String error;
+
+    /// Engine specific details
+    struct KafkaDetails
+    {
+        String topic_name;
+        Int64 partition;
+        Int64 offset;
+        String key;
+    };
+    struct RabbitMQDetails
+    {
+        String exchange_name;
+        String message_id;
+        UInt64 timestamp;
+        bool redelivered;
+        UInt64 delivery_tag;
+        String channel_id;
+    };
+    std::variant<KafkaDetails, RabbitMQDetails> details;
+
+
+    static std::string name() { return "DeadLetter"; }
```

Comment:
Can I ask why it was renamed to `DeadLetter` instead of `DeadLetterQueue`? `system.dead_letter` feels weird, `system.dead_letter_queue` feels more natural, but there might be something that I don't see or understand.

Reply:
I see your point, especially the first one. It is not really a queue in that sense. Let's wait for @morsapaes with final verdict, but I can think of two more names:
1. `dead_letter*s*`: similar to `databases`, `tables`, `parts`
2. `dead_letter_messages`: it is kind of `dead_letter_mails`, but I think `messages` suits this use case more than `mails`.